src/commands.ts

```diff
@@ -637,6 +666,11 @@ export class CommandCenter {
 		await this.commitWithAnyInput(repository);
 	}
 
+	@command('hg.commitAmend', { repository: true })
+	async commitAmend(repository: Repository): Promise<void> {
+		await this.commitWithAnyInput(repository, { amend: true });
```

Comment:
Amend still doesn't work for this use-case. The issue is that `scope` is not defined on the `opts` argument. If you follow this through to the method `commands.smartCommit()` you'll notice that on lines [570-590](https://github.com/incidentist/vscode-hg/blob/b822cf68b848c3024ded50a883f9b09388fa7bea/src/commands.ts#L570) there is code that replaces the `opts` argument entirely. 

A couple of options here. You could change these lines to only overwrite the opts arguments, i.e.
```
opts = { ...opts, scope: CommitScope.STAGED_CHANGES };
```
... or alternatively, you could revert `CommitOptions` values back to being mandatory rather than optional. In this case, you'll need to add something like `CommitScope.AUTOMATIC` to use for `commit` and `commitAmend`.

Reply:
... the way I observed this behaviour was by trying to amend a commit without any files being staged.